share/functions/__fish_git_prompt.fish

```diff
@@ -417,6 +417,14 @@ function __fish_git_prompt --description "Prompt function for Git"
         set u "$___fish_git_prompt_color_untrackedfiles$u$___fish_git_prompt_color_untrackedfiles_done"
     end
     set b (string replace refs/heads/ '' -- $b)
+     if test -z "$__fish_git_prompt_branch_max_length"
+        set __fish_git_prompt_branch_max_length 20
+    end
+    set -l b_ori_length (string length $b)
+    set b (string sub -l $__fish_git_prompt_branch_max_length $b)
+    if test $b_ori_length -gt $__fish_git_prompt_branch_max_length
```

Comment:
This should probably both quote the variable and silence stderr - if it has more than one element or isn't an integer, this `test` will fail.

Which is alright (it should probably do what `$fish_prompt_pwd_dir_length` does in that case and not truncate), but it'll also print an error, which isn't.

Reply:
In general, `test` is the builtin that needs the most care.